spinaltap-mysql/src/main/java/com/airbnb/spinaltap/mysql/schema/MysqlSchemaStore.java

```diff
@@ -5,438 +5,360 @@
-  private static final String CREATE_ARCHIVE_TABLE_QUERY = "CREATE TABLE `%s`.`%s` LIKE `%s`";
-  private static final String INSERT_ARCHIVE_TABLE_QUERY =
-      "INSERT INTO `%s`.`%s` SELECT * FROM `%s` WHERE `database` = '%s'";
-  private static final String DELETE_DATABASE_FROM_SCHEMA_STORE_TABLE_QUERY =
-      "DELETE FROM `%s` WHERE `database` = '%s'";
-
-  private final DBI jdbi;
-  private final String archiveDatabase;
+  private static final String PUT_SCHEMA_QUERY =
+      "INSERT INTO `%s`.`%s`"
+          + " (`database`, `table`, `binlog_file_position`, `server_uuid`, `gtid_set`, `gtid`, `columns`, `sql`, `meta_data`, `timestamp`)"
+          + " VALUES (:database, :table, :binlog_file_position, :server_uuid, :gtid_set, :gtid, :columns, :sql, :meta_data, :timestamp)";
+  private final String sourceName;
+  private final String storeDBName;
+  private final String archiveDBName;
+  private final Jdbi jdbi;
+  private final MysqlSourceMetrics metrics;
+  // Schema cache should always reflect the schema we currently need
+  @Getter
+  private final Table<String, String, MysqlTableSchema> schemaCache =
+      Tables.newCustomTable(Maps.newHashMap(), Maps::newHashMap);
 
-  public MysqlSchemaStore(
-      @NotNull final String source,
-      @NotNull final DBI jdbi,
-      @NotNull final String archiveDatabase,
-      @NotNull final MysqlSourceMetrics metrics) {
-    super(source, metrics);
-    this.jdbi = jdbi;
-    this.archiveDatabase = archiveDatabase;
+  public boolean isCreated() {
+    return jdbi.withHandle(
+            handle ->
+                handle
+                    .createQuery(
+                        "SELECT TABLE_NAME FROM information_schema.tables WHERE table_schema = :db AND table_name = :table")
+                    .bind("db", storeDBName)
+                    .bind("table", sourceName)
+                    .mapTo(String.class)
+                    .findFirst())
+        .isPresent();
   }
 
-  @Override
-  public void put(@NotNull final MysqlTableSchema schema) {
-    try (Handle handle = jdbi.open()) {
-      MysqlSchemaUtil.VOID_RETRYER.call(
-          () -> {
-            handle.insert(
-                String.format(PUT_SCHEMA_QUERY, source),
-                schema.getVersion(),
-                schema.getSource(),
-                schema.getDatabase(),
-                schema.getTable(),
-                schema.getBinlogFilePos().getFileName(),
-                schema.getBinlogFilePos().getPosition(),
-                OBJECT_MAPPER.writeValueAsString(schema));
-            return null;
-          });
-      metrics.schemaStorePutSuccess(schema.getDatabase(), schema.getTable());
-    } catch (Exception ex) {
-      log.error("Failed to put schema {}.", schema);
-      metrics.schemaStorePutFailure(schema.getDatabase(), schema.getTable(), ex);
-      throw new RuntimeException(ex);
+  public void loadSchemaCacheUntil(BinlogFilePos pos) {
```

Comment:
Since `CachedMysqlSchemaStore` is removed, we load table schemas into `schemaCache` here.